runtime/src/commonMain/kotlin/app/cash/sqldelight/db/QueryResult.kt

```diff
@@ -0,0 +1,26 @@
+package app.cash.sqldelight.db
+
+sealed interface QueryResult<T> {
+  val value: T get() = throw IllegalStateException(
+    """
+      The driver used with SQLDelight is asynchronous, so SQLDelight should be configured for
+      asynchronous usage:
+
+      sqldelight {
+        MyDatabase {
+          generateAsync = true
+        }
+      }
+    """.trimIndent()
+  )
+
+  suspend fun await(): T
+
+  data class Value<T>(override val value: T) : QueryResult<T> {
```

Comment:
How about making it an inline class?

Reply:
Inline classes can't have generic properties so that isn't an option here unfortunately.

Reply:
Uff, didn't know this limitation.